src/esn.py

```diff
@@ -96,6 +128,21 @@ def normalize_spectral_radius(self, rho: float = 1.0) -> None:
         current_rho = max(abs(np.linalg.eig(self.w)[0]))
         # scale weight matrix to desired spectral radius
         self.w *= rho / current_rho
+    
+    def normalize_spectral_radius_rfc(self, rho: float = 1.0) -> None:
```

Comment:
this can be part of the existing normalisation function, just add a `is_random_feature` boolean parameter in the function call